test/compare_results.py

```diff
@@ -3,65 +3,118 @@
+            break
 
-def extract_quantum_energies(block):
-    """Extract quantum energies from entire block"""
-    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+    if start_idx is not None:
+        raise Exception(f"{blockname} not found")
+    elif end_idx is not None:
+        raise Exception(f"Could not find end of {blockname}")
+
+    return start_idx, end_idx
 
 def read_chk_file(fname):
     """Read checkpoint file as a list of lines"""
     with open(fname, 'r') as fp:
         # Strip newlines and lines with comments
-        lines = [line for line in fp.readlines() if line != '\n' and '<-' not in line]
+        lines = fp.readlines()
     return lines
 
+def strip_newlines(lines):
+    """Remove all lines which are just newlines"""
+    return [line for line in lines if line != '\n']
+
+def strip_comments(lines):
+    """Remove all lines with comments"""
+    return [line for line in lines if '<-' not in line]
+
+def extract_column(lines, column_no):
+    """Extract column of numbers from list of str lines"""
+    return [float(line.split()[column_no]) for line in lines]
+
 def read_energy_column(fname, column_no):
     """Extract energies from a column in file fname"""
     lines = read_chk_file(fname)
-    lines = lines[:-1] # remove last line (which is not part of the actual data)
-    return [float(line.split()[column_no]) for line in lines]
+    lines = strip_newlines(strip_comments(lines))
+    # remove last line (which is not part of the actual data)
+    lines = lines[:-1]
+    return extract_column(lines, column_no)
 
-def read_quantum_block(fname):
+def read_quantum_energies(fname):
     """Extract quantum energies from fname"""
     lines = read_chk_file(fname)
-    return extract_quantum_block(lines)
+    lines = strip_newlines(strip_comments(lines))
+    start, end = find_start_end_block(lines, "Quantum")
+    # take out first 4 lines and last line of block
+    lines = lines[start+4:end]
+    return extract_column(lines, QUANTUM_ENERGY_IDX)
 
-def compare_columns(fname1, fname2, column_no, precision=1e-10):
+def read_intensity_column(fname, column_name):
+    """Extract quantum energies from fname"""
+    if column_name not in INTENSITY_INDICES.keys():
+        raise Exception(f"Intensity column name must be one of {INTENSITY_INDICES.keys()}")
+
+    lines = read_chk_file(fname)
+    lines = strip_newlines(lines)
+    start, end = find_log_block(lines, "Linestrength")
+    # take out first 4 lines and last line of block
+    lines = lines[start+1:end]
+
+    return extract_column(lines, INTENSITY_INDICES[column_name])
+
+def compare_columns(col1, col2, abs_precision=0.0, rel_precision=1e-10):
+    """Compare two columns of numbers to a give absolute or relative precision"""
+    for i, (e1, e2) in enumerate(zip(col1, col2)):
+        if not math.isclose(e1, e2, abs_tol=abs_precision, rel_tol=rel_precision):
+            print(f"{e1} and {e2} differ by {abs(e1-e2)} at index {i}")
```

Comment:
Will this go back to an assertion/exception when we're happy with it? Or do we we have another way of detecting failure (or don't need one)?